.github/workflows/docker.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: Docker
+
+on:
+  push:
+    tags: [ '*' ]
+
+jobs:
+  docker:
+    strategy:
+      matrix:
+        python_base_version:
+          - 3.9
+          - 3.10
+          - 3.11
+        ansible_version:
+          - 6.7.0
+          - 7.2.0
+        ansible_lint_version:
+          - 6.13.1
+        exclude:
+          python_base_version: 3.11
+          ansible_version: 6.7.0
+
+    uses: famedly/github-workflows/.github/workflows/docker.yml@docker-v1
+    with:
+      push: ${{ github.event_name != 'pull_request' }} # Always build, don't publish on pull requests
+      registry_user: famedly_ci
+      registry: docker-oss.nexus.famedly.de
+      image_name: ansible
+      build_args: "PYTHON_BASE_VERSION=${{ matrix.python_base_version }}, ANSIBLE_VERSION=${{ matrix.ansible_version }}, ANSIBLE_LINT_VERSION=${{ matrix.ansible_lint_version }}" 
+    secrets: inherit
```

Comment:
after what we had with the powerdns image, I'd assume we'll be running into the same issue here, right? The merged image will contain the images from every job in this definition, not just the ones that are part of the same matrix variant, right?

The problem was that the digests that are uploaded as artifacts aren't scoped in any way, and I don't see how we're avoiding the problem here.

Aside of that: With GitLab, we ended up having `ansible:py<version>-ansible<version>` images, where as here we'd not get that *yet*, right? Sorry if I'm missing something here

Reply:
No, you're not missing anything, I'm still working on getting this right atm, I should have switched the PR to a Draft so it doesn't automatically send out a request review after pushing ^^

Reply:
Ah, okay. Thanks for switching it to a draft then :)

Reply:
> The merged image will contain the images from every job in this definition, not just the ones that are part of the same matrix variant, right?

I'm not sure, but I don' think so, since the merge is also called once per matrix variant, and the digest is specific to each build, as long as we implement the different tags as `ansible:py<version>-ansible<version>` it should only merge the ones that belong together? Please do tell if I'm misunderstanding something still, it's a bit complex and hard to wrap my head around without actually having done any test runs yet

Reply:
With two different workflow invocations for different container registries, we still had the problem in the powerdns case.

The problem is that these two bits aren't isolated from other imported workflows running in the same workflow:

https://github.com/famedly/github-workflows/blob/main/.github/workflows/docker.yml#L92-L104 and https://github.com/famedly/github-workflows/blob/main/.github/workflows/docker.yml#L113-L117

In the powerdns case we were building 4 images: nexus arm, nexus x86, ghcr arm and ghcr x86. The merge job failed in the nexus case because it couldn't pull the images pushed to ghcr, and the merge job for ghcr failed because it couldn't find the images pushed to nexus.

Without this isolation, I'd currently assume that the matrix build will run into the same issue: Even though we have multiple merge jobs, each of the merge jobs will collect all of the built images, even when they shouldn't.

afaict it should be possible do do this by adding `${{ inputs.registry }}/${{ inputs.image_name }}/${{ inputs.tag }}` or something to the artifact path, which would ensure that these artifacts don't clash.

Reply:
Now that https://github.com/famedly/github-workflows/pull/16 is merged this should also be ready for a merge, right?

Reply:
I'd like to rerun the workflow and verify that the inspect things look good, but after that's done, yeah

Reply:
I have triggered the re-run

Reply:
The re-run isn't using the updated reference to the workflows repo, it's still using the specific commit hash from earlier